packages/experimental/Link/src/useLink.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import {
+  useAsPressable,
+  useKeyProps,
+  useOnPressWithFocus,
+  useViewCommandFocus,
+  getAccessibilityState,
+} from '@fluentui-react-native/interactive-hooks';
+import { LinkProps, LinkState } from './Link.types';
+import { Linking } from 'react-native';
+
+export const useLink = (props: LinkProps): LinkState => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    accessible = true,
+    accessibilityRole,
+    onPress,
+    onAccessibilityTap,
+    tooltip,
+    url,
+    accessibilityState,
+    componentRef = defaultComponentRef,
+    disabled,
+    enableFocusRing,
+    focusable,
+    ...rest
+  } = props;
+  const isDisabled = !!disabled;
+
+  const [visitedState, setVisitedState] = React.useState(false);
+  const linkOnPress = React.useCallback(
+    (e) => {
+      setVisitedState(true);
+      if (url) {
+        Linking.openURL(url as string);
+      } else if (onPress) {
+        onPress(e);
+      }
+      e.stopPropagation();
+    },
+    [setVisitedState, url, onPress],
+  );
+
+  // GH #1336: Set focusRef to null if link is disabled to prevent getting keyboard focus.
+  const focusRef = isDisabled && !focusable ? null : componentRef;
```

Comment:
Probably worth separating this out into its own hook too, since the logic just keeps being reused. Not blocking.